### ISSUE TITLE
Error when loading CSM mod

### ISSUE THREAD

**Issue by EvolutionPixels** [minetest/minetest#9091]
![изображение](https://user-images.githubusercontent.com/12032252/68457233-dbe26100-0210-11ea-95ad-6dc8b175d0c7.png)

This error appears in mod [Location markers](https://github.com/matyilona/locations_markers_csm) when loading the server. The reason isn't the mod themselves, I guess.

Minetest 5.1.0 release.

**Comment by ClobberXD** [comment]
> locations_markers_csm:init.lua:28: attempt to index upvalue 'player' (a nil value)

This does indeed seem to be a mod error. The error message even pinpoints the erroneous line in the script.

**Comment by EvolutionPixels** [comment]
> 
> 
> > locations_markers_csm:init.lua:28: attempt to index upvalue 'player' (a nil value)
> 
> This is a mod error. The error message even pinpoints the erroneous line in the script.

The problem is that there's no way to fix this error. Lua API for CSM seems to have been broken.

**Comment by sfan5** [comment]
The mod expects `minetest.localplayer` to exist at loading time, the engine does not promise that this is the case.
Unrelated to that, the last commit that touched this is 23677be951b4dc2b9cc174ceb92f0e0b8e8eb867.

**Comment by EvolutionPixels** [comment]
> 
> 
> The mod expects `minetest.localplayer` to exist at loading time, the engine does not promise that this is the case.
> Unrelated to that, the last commit that touched this is [23677be](https://github.com/minetest/minetest/commit/23677be951b4dc2b9cc174ceb92f0e0b8e8eb867).

The error is still present in the latest version 5.2.0 dev as in the screenshot above.

**Comment by LoneWolfHT** [comment]
IIRC this is due to minetest.localplayer not existing when CSMs are disabled server-side

**Comment by EvolutionPixels** [comment]
> 
> 
> IIRC this is due to minetest.localplayer not existing when CSMs are disabled server-side

Thank you for some explanation for the problem.

**Comment by nerzhul** [comment]
the mod should check if localplayer exists and then disable some code parts if not

**Comment by jastevenson303** [comment]
I ran into this too, but didn't realize it was changed behavior.
Previously I had made this mod
https://github.com/jastevenson303/sjcsm/blob/master/init.https://forum.minetest.net/viewtopic.php?f=53&t=23607#L9
using
minetest.localplayer as you do in location_markers.

I had been doing it like this
https://github.com/jastevenson303/ll_csm_jumpsound/blob/master/init.lua#L36
recently
without noticing the changed behavior.  Checking if player exists and
setting it to minetest.localplayer if not.

Thanks for the issue; I didn't notice the changed behavior.  There is a
workaround here, but I am curious when the change occurred.  sfan5 had a
commit for testing... It's not a problem of CSM disabled/enabled
server-side however, Lone_Wolf.
